### DISCUSSION TITLE
URL rewrite logic

### DISCUSSION THREAD

**Discussion by arkadiuszwojcik** [OrchardCMS/OrchardCore#16629] [Ideas]
I would like to ask you about url rewrite funtionality in OC. I know that there is already request for such feature: #9027 and in one of the comments @Piedone suggests to use [URL Rewriting Middleware](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-8.0). As I noticed `RewriteOptions` object allow to accept apache mod rules by using `AddApacheModRewrite` this allow to create very flexible solution with just simple text box settings page in OC admin dashboard. Can somone points out what might be potential pitfalls of such approach? For example one of my concer is how to make sure such config will not affect other tenents? I guess I will need some extra validation / restrictions? Are there any other potential issues? I'm curious about your opinion.

**Comment by Piedone** [comment]
You correctly point out that such rules would affect other tenants, so that's indeed an issue. I don't know how best to match URL Rewriting with OC's tenant lookup, because it's not just about registering dependencies in the DI container.

Also, performance is something to consider, since URL rewriting rules are evaluated for every request. I'm sure this is optimized in URL Rewriting, but it can affect your site.

**Reply by arkadiuszwojcik** [reply]
I made some tests, looks like `UseRewriter` with `RewriteOptions` adds middleware that affects only given tenant (as each tenant have its own `IApplicationBuilder`) so this is not an issue.

**Reply by Piedone** [reply]
Ah, awesome!
